docs(framework): Introduce migration guide for `Message API` (#5770)
adap/flower: framework/docs/source/how-to-upgrade-to-flower-with-message-api.rst

.. |numpyclient_link| replace:: ``NumPyClient``

.. _numpyclient_link: ref-api/flwr.client.NumPyClient.html

.. |client_link| replace:: ``Client``

.. _client_link: ref-api/flwr.client.Client.html

.. |clientapp_link| replace:: ``ClientApp``

.. _clientapp_link: ref-api/flwr.client.ClientApp.html

.. |serverapp_link| replace:: ``ServerApp``

.. _serverapp_link: ref-api/flwr.server.ServerApp.html

.. |strategy_link| replace:: ``Strategy``

.. _strategy_link: ref-api/flwr.serverapp.Strategy.html

.. |fedavg_link| replace:: ``FedAvg``

.. _fedavg_link: ref-api/flwr.serverapp.FedAvg.html

.. |message_link| replace:: ``Message``

.. _message_link: ref-api/flwr.common.Message.html

.. |arrayrecord_link| replace:: ``ArrayRecord``

.. _arrayrecord_link: ref-api/flwr.common.ArrayRecord.html

.. |metricrecord_link| replace:: ``MetricRecord``

.. _metricrecord_link: ref-api/flwr.common.MetricRecord.html

.. |configrecord_link| replace:: ``ConfigRecord``

.. _configrecord_link: ref-api/flwr.common.ConfigRecord.html

.. |recorddict_link| replace:: ``RecordDict``

.. _recorddict_link: ref-api/flwr.common.RecordDict.html

> Upgrade to the Message API
> ==========================

Reviewer: ```suggestion
Upgrade to Message API
======================
```

> Welcome to the migration guide for updating your FlowerApps to use Flower's Message API!

Reviewer: Always `Flower App`, not `FlowerApp`:
```suggestion
Welcome to the migration guide for updating your Flower Apps to use Flower's Message API!
```

> This guide will walk you through the necessary steps to transition from Flower Apps
> based on Strategies and NumPyClient to their equivalent using the new Message API. This
> guide is relevant when updating pre-``1.21`` Flower apps to the latest stable version.

Reviewer: ```suggestion
This guide will walk you through the necessary steps to transition from Flower Apps
based on ``Strategy`` and ``NumPyClient`` to their equivalent using the new Message API. This
guide is relevant when updating pre-``1.21`` Flower apps to the latest stable version.
```
